lib/measured/measurable.rb

```diff
@@ -11,10 +11,12 @@ def initialize(value, unit)
     raise Measured::UnitError, "Unit #{ unit } does not exist" unless self.class.conversion.unit_or_alias?(unit)
 
     @value = case value
-    when NilClass
-      raise Measured::UnitError, "Unit value cannot be nil"
     when Float
       BigDecimal(value, Float::DIG+1)
+    when Fixnum, Bignum
+      BigDecimal(value)
+    when NilClass
+      raise Measured::UnitError, "Unit value cannot be nil"
```

Comment:
Added a check for `Fixnum` and `Bignum`, plus rearranged things for the common case.